_source/_posts/2022-07-06-spa-web-security.md

```diff
@@ -0,0 +1,137 @@
+
+So the advice is to enable CORS appropriately with considered use of allowlists and to use proxies as appropriate (such as for local development). The concept of least privilege applies here too, and keeping a tight lid on things helps mitigate security vulnerabilities.
+
+### Dive into cross-origin resource sharing
+Want to understand better when CORS rules apply or how to set up the allowlist rules? Read these resources for a deep dive.
+* [MDN CORS Reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+* [Auth0's CORS Tutorial](https://auth0.com/blog/cors-tutorial-a-guide-to-cross-origin-resource-sharing/)
+
+## Protect your cookies from theft
+Really, this advice is applicable outside of web development too. But since we're talking about web development in this post, we'll skip mitigation strategies about protecting your physical cookies from marauders and focus again on security headers.
+
+Cookies can contain all sorts of goodies, such as chocolate chips and the tidbits of data that agitators love to have! Current authentication best practices favor using in-browser memory or web workers for authentication tokens over cookies, and cookies should never be used for secrets, but there might still be sensitive information stored in cookies. To help protect cookies, we can use attributes.
+
+Cookies should be temporary. Particularly cookies with sensitive information should be as short-lived as possible. When you or your back-end bakes up a cookie, adding the attribute `Max-Age` defines the longevity of the cookie. 
+
+There are some more attributes you can bake in! Adding the `httpOnly` attribute means the browser only sends the cookie for HTTP requests, and JavaScript won't be able to access cookies. Setting the `httpOnly` attribute is an excellent first step to guarding your cookies against a malicious script that reads your cookies.
+
+Having the `httpOnly` attribute means that JavaScript can't access the cookie, but it doesn't mean your browser carefully curates **which** cookies to send to each HTTP request! It just sends them all, resulting in a possibility of a vulnerability!
+
+This is where the `SameSite` attribute comes into play. You can control when cookies should be sent by using the `SameSite` attribute and setting one of 3 values:
+1. `Strict` - only send cookies if it's going to the same site
+2. `Lax` - only send cookies when the user is navigating to the origin site (this is the default behavior for Chromium-based browsers now)
+3. `None` - send the cookies to everyone, everywhere. As a safeguard for your concerningly generous behavior, you're also required to add the `Secure` attribute
```

Comment:
```suggestion
3. `None` - send the cookies to everyone, everywhere. As a safeguard for your generous behavior, which concerns us, you're also required to add the `Secure` attribute.
```
clarity

Reply:
@havi will this work?

3. `None` - send the cookies to everyone, everywhere. As a safeguard for your generous behavior, which is concerning to the browser, you're also required to add the `Secure` attribute

Reply:
WFH! thanks :)